thunder/core/pytree.py

```diff
@@ -21,7 +21,15 @@
 )
 
 
-def tree_flatten(args, namespace=""):
+optree.register_pytree_node(
+    slice,
+    lambda s: ([s.start, s.stop, s.step], None, None),
+    lambda _, children: slice(*children),
+    namespace=OPTREE_NAMESPACE,
+)
+
+
+def tree_flatten(args, namespace=OPTREE_NAMESPACE):
```

Comment:
following the earlier question, when would we want to avoid flattening without custom node?